DuckDuckGo/MainWindow/MainWindowController.swift

```diff
@@ -67,6 +71,10 @@ final class MainWindowController: NSWindowController {
 
     deinit {
         NotificationCenter.default.removeObserver(self)
+
+        if #available(macOS 14.4, *) {
+            WebExtensionManager.shared.eventsListener.didCloseWindow(self)
```

Comment:
it‘s probably a wrong place to call the notification passing the object being deallocated that most probably will lead to a crash.
I think we should use `windowWillClose` handler or something like it.

Reply:
yeah good point 👍